blah.cpp

```diff
@@ -0,0 +1,82 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <iostream>
+#include <cstring>
+#include <cstdlib>
+#include <cctype.h>
+
+bool characteristic(char numString[], int& c);
+bool mantissa(char numString[], int& numerator, int& denominator);
+
+int main()
+{
+    char number[] = "123.456";
+    int c, n, d;
+
+    //if the conversion from C string to integers can take place
+    if(characteristic(number, c) && mantissa(number, n, d))
+    {
+        //do some math with c, n, and d
+        printf("test true \n");
+        printf("The characteristic is: %i\n", c);
+        printf("The mantissa numerator is: %i\n", n);

```

Comment:
I originally thought printf() shouldn't work in C++, since it's C, but it turns out I'm wrong. I would personally advocate for using std::cout just because this is C++, and you aren't outputting enough IO for the performance issue with cout to matter. That being said, here the interesting StackOverflow thread that showed me this. https://stackoverflow.com/questions/2872543/printf-vs-cout-in-c